### PR TITLE
uncommented school controller tests now all trams API code is in

### PR DESCRIPTION
**Pull request by MarkRobinsonDfe**

### CODE REVIEW

**Review by wystanpalm-madetech — CHANGES_REQUESTED** [review]
Please rebase on master once #66 is in

File: Dfe.Academies.External.Web.UnitTest/Controllers/SchoolControllerTests.cs [hunk 1]
```diff
@@ -3,59 +3,84 @@
...
 [Parallelizable(ParallelScope.All)]
 internal sealed class SchoolControllerTests
 {
-    /// <summary>
-    /// TODO MR:- re-test after PR55 - plumb in trams API
-    /// </summary>
-    //[Test]
-    public async Task SchoolController___Search___ReturnsResult()
+    [Test]
+    public async Task Seach___ResultsFound___ResultsReturned()
```
Comment (CathLass) [inline Dfe.Academies.External.Web.UnitTest/Controllers/SchoolControllerTests.cs:22]: ReferenceDataRetrievalService.SearchSchools(schoolSearch) returns null in this test

File: Dfe.Academies.External.Web.UnitTest/Controllers/SchoolControllerTests.cs [hunk 2]
```diff
@@ -3,59 +3,84 @@
...
-        //Assert.That(viewModel, Is.Not.Null);
     }
 
-    /// <summary>
-    /// TODO MR:- re-test after PR55 - plumb in trams API
-    /// </summary>
-    //[Test]
-    public async Task SchoolController___ReturnSchoolDetailsPartialViewPopulated___ReturnsPartialView()
+    [Test]
+    public async Task ReturnSchoolDetailsPartialViewPopulated___ValidSchool___ReturnsPartialView()
```
Comment (CathLass) [inline Dfe.Academies.External.Web.UnitTest/Controllers/SchoolControllerTests.cs:45]: result is a 500 error in this test

**Review by CathLass — CHANGES_REQUESTED** [review]

File: Dfe.Academies.External.Web.UnitTest/Controllers/SchoolControllerTests.cs (hunk 1 above)
Comment (MarkRobinsonDfe) [reply Dfe.Academies.External.Web.UnitTest/Controllers/SchoolControllerTests.cs]: plumbed in mock IHttpClientFactory which has fixed this

File: Dfe.Academies.External.Web.UnitTest/Controllers/SchoolControllerTests.cs (hunk 2 above)
Comment (MarkRobinsonDfe) [reply Dfe.Academies.External.Web.UnitTest/Controllers/SchoolControllerTests.cs:53]: plumbed in mock IHttpClientFactory which has fixed this

**Review by CathLass — APPROVED** [review]

**Review by wystanpalm-madetech — APPROVED** [review]

**Pull request merged**